### PR TITLE
Add NegativeContainsPattern (aka notContaining) (#1881)

### PR DESCRIPTION
**Pull request by dorzepowski**
Hi, found this issue https://github.com/wiremock/wiremock/issues/1881 as [good first issue](https://github.com/wiremock/wiremock/labels/good%20first%20issue) during [Hacktoberfest](https://hacktoberfest.com/).

### CODE REVIEW

File: src/main/java/com/github/tomakehurst/wiremock/matching/StringValuePatternJsonDeserializer.java [hunk 1]
```diff
@@ -49,6 +49,7 @@ public class StringValuePatternJsonDeserializer extends JsonDeserializer<StringV
           .put("equalToXml", EqualToXmlPattern.class)
           .put("matchesXPath", MatchesXPathPattern.class)
           .put("contains", ContainsPattern.class)
+          .put("doesNotContains", NegativeContainsPattern.class)
```
Comment (tomakehurst) [inline src/main/java/com/github/tomakehurst/wiremock/matching/StringValuePatternJsonDeserializer.java:52]: This should be `doesNotContain` (i.e. not plural)

File: src/test/java/com/github/tomakehurst/wiremock/StubbingAcceptanceTest.java [hunk 2]
```diff
@@ -668,6 +696,27 @@ public void matchingOnMultipartRequestBodyWithAContainsAndANegativeRegex() {
     assertThat(response.statusCode(), is(HTTP_OK));
   }
 
+  @Test
+  public void matchingOnMultipartRequestBodyWithNotContaining() {
+    stubFor(
+        post(urlEqualTo("/match/this/part/too"))
+            .withMultipartRequestBody(
+                aMultipart()
+                    .withHeader("Content-Type", containing("text/plain"))
```
Comment (tomakehurst) [inline src/test/java/com/github/tomakehurst/wiremock/StubbingAcceptanceTest.java:705]: I presume you meant to test the `notContaining` matcher here?

File: src/test/java/com/github/tomakehurst/wiremock/matching/NegativeContainsPatternTest.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.github.tomakehurst.wiremock.client.WireMock;
+import org.junit.jupiter.api.Test;
+
+public class NegativeContainsPatternTest {
```
Comment (tomakehurst) [inline src/test/java/com/github/tomakehurst/wiremock/matching/NegativeContainsPatternTest.java:26]: A 3rd test case would be good showing what happens when the actual value is null, as can happen when e.g. the request header being compared is absent.

**Review by tomakehurst — CHANGES_REQUESTED** [review]

File: src/test/java/com/github/tomakehurst/wiremock/StubbingAcceptanceTest.java (hunk 2 above)
Comment (dorzepowski) [reply src/test/java/com/github/tomakehurst/wiremock/StubbingAcceptanceTest.java]: Good point. Thanks.

File: src/test/java/com/github/tomakehurst/wiremock/matching/NegativeContainsPatternTest.java (hunk 3 above)
Comment (dorzepowski) [reply src/test/java/com/github/tomakehurst/wiremock/matching/NegativeContainsPatternTest.java:26]: Yes, good point. I don't know why I forgot about adding this.

Thank you for all those comments. I don't know what are the contribution rules for commits, therefore I just amend my commit with fixes applied to it.

**Comment by tomakehurst** [comment]
Thanks!

**Comment by hickford** [comment]
"doesNotContain" missing documentation at https://docs.wiremock.io/request-matching/matcher-types

**Pull request merged**